src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp

```diff
@@ -70,13 +70,7 @@ class UavcanDynamicPortSubscriber : public UavcanBaseSubscriber
 			uavcan_register_Value_1_0 value;
 
 			if (_param_manager.GetParamByName(uavcan_param, value)) {
-				int32_t new_id = value.integer32.value.elements[0];
-
-				// Allow, for example, a default PX4 param value of '-1' to disable subscription
-				if (!isValidPortId(new_id)) {
-					// but always use the standard 'unset' value for comparison
-					new_id = CANARD_PORT_ID_UNSET;
-				}
+				uint16_t new_id = value.natural16.value.elements[0];
```

Comment:
Don't we still need a check after here to ensure that `new_id <= CANARD_PORT_ID_MAX`?

Reply:
The callback return a boolean, my assumption would be that if it's returning true it means it got a valid port_id >=0 and <= CANARD_PORT_ID_MAX

Reply:
Oh, I see. 👍🏻

Reply:
I think this is ready for merge now.